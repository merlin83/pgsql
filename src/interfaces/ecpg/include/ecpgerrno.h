#ifndef _ECPG_ERROR_H
#define _ECPG_ERROR_H

#include <errno.h>

/* This is a list of all error codes the embedded SQL program can return */
#define	ECPG_NO_ERROR		0
#define ECPG_NOT_FOUND		100

/* system error codes returned by ecpglib get the correct number,
 * but are made negative
 */
#define ECPG_OUT_OF_MEMORY	-ENOMEM

/* first we have a set of ecpg messages, they start at 200 */
#define ECPG_UNSUPPORTED	-200
#define ECPG_TOO_MANY_ARGUMENTS	-201
#define ECPG_TOO_FEW_ARGUMENTS	-202
#define ECPG_TOO_MANY_MATCHES	-203
#define ECPG_INT_FORMAT		-204
#define ECPG_UINT_FORMAT	-205
#define ECPG_FLOAT_FORMAT	-206
#define ECPG_CONVERT_BOOL	-207
#define ECPG_EMPTY		-208
#define ECPG_NO_CONN		-209

/* finally the backend error messages, they start at 300 */
#define ECPG_PGSQL		-300
#define ECPG_TRANS		-301
#define ECPG_CONNECT		-302

#endif /* !_ECPG_ERROR_H */
