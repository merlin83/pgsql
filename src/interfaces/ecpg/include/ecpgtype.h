/*
 * This file implements a data structure that is built and maintained by the
 * preprocessor.
 *
 * All types that can be handled for host variable declarations has to
 * be handled eventually.
 */

/*
 * Here are all the types that we are to handle. Note that it is the type
 * that is registered and that has nothing whatsoever to do with the storage
 * class.
 *
 * Simle types
 * integers: char, short, int, long (signed and unsigned)
 * floats: float, double
 *
 * Complex types:
 * VARCHAR, VARCHAR2 - Strings with length (maxlen is given in the declaration)
 * Arrays of simple types and of VARCHAR, VARCHAR2 (size given in declaration)
 * Records build of simple types, arrays and other records.
 *
 * Complicating things:
 * typedefs and struct names!
 *
 * Conclusion:
 * This is a typically recursive definition. A structure of typed list elements
 * would probably work fine:
 */
#include <stdio.h>

enum ECPGttype {
    ECPGt_char = 1, ECPGt_unsigned_char, ECPGt_short, ECPGt_unsigned_short, 
    ECPGt_int, ECPGt_unsigned_int, ECPGt_long, ECPGt_unsigned_long,
    ECPGt_bool,
    ECPGt_float, ECPGt_double,
    ECPGt_varchar, ECPGt_varchar2,
    ECPGt_array,
    ECPGt_record,
    ECPGt_EOIT,			/* End of insert types. */
    ECPGt_EORT			/* End of result types. */
};

#define IS_SIMPLE_TYPE(type) ((type) >= ECPGt_char && (type) <= ECPGt_varchar2)
