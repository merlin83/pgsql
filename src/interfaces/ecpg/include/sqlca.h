#ifndef POSTGRES_SQLCA_H
#define POSTGRES_SQLCA_H

#ifdef __cplusplus
extern "C" {
#endif

struct sqlca
{
	int			sqlcode;
	struct
	{
		int			sqlerrml;
		char		sqlerrmc[1000];
	}			sqlerrm;
}			sqlca;

#endif

#ifdef __cplusplus
}
#endif
 
