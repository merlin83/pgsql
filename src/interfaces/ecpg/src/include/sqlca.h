#ifndef POSTGRES_SQLCA_H
#define POSTGRES_SQLCA_H

struct sqlca {
    int sqlcode;
    struct {
	int sqlerrml;
	char sqlerrmc[1000];
    } sqlerrm;
} sqlca;
#endif
