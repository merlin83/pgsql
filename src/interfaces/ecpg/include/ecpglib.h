#include <postgres.h>

#ifdef __cplusplus
extern		"C"
{
#endif

	void		ECPGdebug(int, FILE *);
	bool		ECPGstatus(int, const char *);
	bool		ECPGsetconn(int, const char *);
	bool		ECPGconnect(int, const char *, const char *, const char *, const char *);
	bool		ECPGdo(int, const char *, char *,...);
	bool		ECPGtrans(int, const char *, const char *);
	bool		ECPGdisconnect(int, const char *);
	bool		ECPGprepare(int, char *, char *);
	bool		ECPGdeallocate(int, char *);
	char		*ECPGprepared_statement(char *);
		
	void		ECPGlog(const char *format,...);

#ifdef LIBPQ_FE_H
	bool		ECPGsetdb(PGconn *);

#endif

/* Here are some methods used by the lib. */
/* Returns a pointer to a string containing a simple type name. */
	const char *ECPGtype_name(enum ECPGttype);

/* A generic varchar type. */
	struct ECPGgeneric_varchar
	{
		int			len;
		char		arr[1];
	};

/* print an error message */
	void		sqlprint(void);

	struct cursor
	{
		const char *name;
		char	   *command;
		struct cursor *next;
	};

	extern int	no_auto_trans;

/* define this for simplicity as well as compatibility */

#define		  SQLCODE	 sqlca.sqlcode

#ifdef __cplusplus
}

#endif

#include <ecpgerrno.h>
