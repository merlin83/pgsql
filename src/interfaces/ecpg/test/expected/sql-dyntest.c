/* Processed by ecpg (4.2.1) */
/* These include files are added by the preprocessor */
#include <ecpgtype.h>
#include <ecpglib.h>
#include <ecpgerrno.h>
#include <sqlca.h>
/* End of automatic include section */

#line 1 "dyntest.pgc"
/* dynamic SQL test program
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>


#line 1 "./../../include/sql3types.h"
#ifndef _ECPG_SQL3TYPES_H
#define _ECPG_SQL3TYPES_H

/* SQL3 dynamic type codes */

/* chapter 13.1 table 2: Codes used for SQL data types in Dynamic SQL */

enum
{
	SQL3_CHARACTER = 1,
	SQL3_NUMERIC,
	SQL3_DECIMAL,
	SQL3_INTEGER,
	SQL3_SMALLINT,
	SQL3_FLOAT,
	SQL3_REAL,
	SQL3_DOUBLE_PRECISION,
	SQL3_DATE_TIME_TIMESTAMP,
	SQL3_INTERVAL,				/* 10 */
	SQL3_CHARACTER_VARYING = 12,
	SQL3_ENUMERATED,
	SQL3_BIT,
	SQL3_BIT_VARYING,
	SQL3_BOOLEAN,
	SQL3_abstract
	/* the rest is xLOB stuff */
};

/* chapter 13.1 table 3: Codes associated with datetime data types in Dynamic SQL */

enum
{
	SQL3_DDT_DATE = 1,
	SQL3_DDT_TIME,
	SQL3_DDT_TIMESTAMP,
	SQL3_DDT_TIME_WITH_TIME_ZONE,
	SQL3_DDT_TIMESTAMP_WITH_TIME_ZONE,

	SQL3_DDT_ILLEGAL			/* not a datetime data type (not part of
								 * standard) */
};

#endif   /* !_ECPG_SQL3TYPES_H */

#line 8 "dyntest.pgc"


#line 1 "./../../include/sqlca.h"
#ifndef POSTGRES_SQLCA_H
#define POSTGRES_SQLCA_H

#ifndef DLLIMPORT
#if  defined(WIN32) || defined(__CYGWIN__)
#define DLLIMPORT __declspec (dllimport)
#else
#define DLLIMPORT
#endif   /* __CYGWIN__ */
#endif   /* DLLIMPORT */

#define SQLERRMC_LEN	150

#ifdef __cplusplus
extern		"C"
{
#endif

struct sqlca_t
{
	char		sqlcaid[8];
	long		sqlabc;
	long		sqlcode;
	struct
	{
		int			sqlerrml;
		char		sqlerrmc[SQLERRMC_LEN];
	}			sqlerrm;
	char		sqlerrp[8];
	long		sqlerrd[6];
	/* Element 0: empty						*/
	/* 1: OID of processed tuple if applicable			*/
	/* 2: number of rows processed				*/
	/* after an INSERT, UPDATE or				*/
	/* DELETE statement					*/
	/* 3: empty						*/
	/* 4: empty						*/
	/* 5: empty						*/
	char		sqlwarn[8];
	/* Element 0: set to 'W' if at least one other is 'W'	*/
	/* 1: if 'W' at least one character string		*/
	/* value was truncated when it was			*/
	/* stored into a host variable.				*/

	/*
	 * 2: if 'W' a (hopefully) non-fatal notice occurred
	 */	/* 3: empty */
	/* 4: empty						*/
	/* 5: empty						*/
	/* 6: empty						*/
	/* 7: empty						*/

	char		sqlstate[5];
};

struct sqlca_t *ECPGget_sqlca(void);

#ifndef POSTGRES_ECPG_INTERNAL
#define sqlca (*ECPGget_sqlca())
#endif

#ifdef __cplusplus
}
#endif

#endif

#line 9 "dyntest.pgc"


#line 1 "./../regression.h"






#line 10 "dyntest.pgc"


static void error(void)
{  printf("#%ld:%s\n",sqlca.sqlcode,sqlca.sqlerrm.sqlerrmc);
   exit(1);
}

int main(int argc,char **argv)
{ /* exec sql begin declare section */
   
   
   
   
   
   
   
   
   
   
   
   
  
#line 19 "dyntest.pgc"
 int  COUNT    ;
 
#line 20 "dyntest.pgc"
 int  INTVAR    ;
 
#line 21 "dyntest.pgc"
 int  INDEX    ;
 
#line 22 "dyntest.pgc"
 int  INDICATOR    ;
 
#line 23 "dyntest.pgc"
 bool  BOOLVAR    ;
 
#line 24 "dyntest.pgc"
 int  TYPE    ,  LENGTH    ,  OCTET_LENGTH    ,  PRECISION    ,  SCALE    ,  RETURNED_OCTET_LENGTH    ;
 
#line 25 "dyntest.pgc"
 int  DATETIME_INTERVAL_CODE    ;
 
#line 26 "dyntest.pgc"
 char  NAME [ 120 ]    ;
 
#line 27 "dyntest.pgc"
 char  STRINGVAR [ 1024 ]    ;
 
#line 28 "dyntest.pgc"
 float  FLOATVAR    ;
 
#line 29 "dyntest.pgc"
 double  DOUBLEVAR    ;
 
#line 30 "dyntest.pgc"
 char * QUERY    ;
/* exec sql end declare section */
#line 31 "dyntest.pgc"

  int done=0;

  ECPGdebug(1, stderr);

  QUERY="select rulename, ev_class, ev_attr, ev_type, is_instead, ev_qual from pg_rewrite";

  /* exec sql whenever sqlerror  do error (  ) ; */
#line 38 "dyntest.pgc"


  ECPGallocate_desc(__LINE__, "MYDESC");
#line 40 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );
#line 40 "dyntest.pgc"


  { ECPGconnect(__LINE__, 0, "regress1" , NULL,NULL , NULL, 0); 
#line 42 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 42 "dyntest.pgc"


  { ECPGprepare(__LINE__, "MYQUERY" , QUERY);
#line 44 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 44 "dyntest.pgc"

  /* declare MYCURS  cursor  for ? */
#line 45 "dyntest.pgc"


  { ECPGdo(__LINE__, 0, 1, NULL, "declare MYCURS  cursor  for ?", 
	ECPGt_char_variable,(ECPGprepared_statement("MYQUERY")),(long)1,(long)1,(1)*sizeof(char), 
	ECPGt_NO_INDICATOR, NULL , 0L, 0L, 0L, ECPGt_EOIT, ECPGt_EORT);
#line 47 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 47 "dyntest.pgc"


  while (1)
  {
     { ECPGdo(__LINE__, 0, 1, NULL, "fetch in MYCURS", ECPGt_EOIT, 
	ECPGt_descriptor, "MYDESC", 0L, 0L, 0L, 
	ECPGt_NO_INDICATOR, NULL , 0L, 0L, 0L, ECPGt_EORT);
#line 51 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 51 "dyntest.pgc"


     if (sqlca.sqlcode) break;

     { ECPGget_desc_header(__LINE__, "MYDESC", &(COUNT));

#line 55 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 55 "dyntest.pgc"

     if (!done)
     {
        printf("%d Columns\n",COUNT);
        for (INDEX=1;INDEX<=COUNT;++INDEX)
        {
		{ ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_ret_octet,
	ECPGt_int,&(RETURNED_OCTET_LENGTH),(long)1,(long)1,sizeof(int), ECPGd_name,
	ECPGt_char,(NAME),(long)120,(long)1,(120)*sizeof(char), ECPGd_scale,
	ECPGt_int,&(SCALE),(long)1,(long)1,sizeof(int), ECPGd_precision,
	ECPGt_int,&(PRECISION),(long)1,(long)1,sizeof(int), ECPGd_octet,
	ECPGt_int,&(OCTET_LENGTH),(long)1,(long)1,sizeof(int), ECPGd_length,
	ECPGt_int,&(LENGTH),(long)1,(long)1,sizeof(int), ECPGd_type,
	ECPGt_int,&(TYPE),(long)1,(long)1,sizeof(int), ECPGd_EODT);

#line 66 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 66 "dyntest.pgc"

			printf("%s ",NAME);
        	switch (TYPE)
        	{  case SQL3_BOOLEAN:
        			printf("bool ");
        			break;
        	   case SQL3_NUMERIC:
        	   		printf("numeric(%d,%d) ",PRECISION,SCALE);
        	        break;
        	   case SQL3_DECIMAL:
        	   		printf("decimal(%d,%d) ",PRECISION,SCALE);
        	        break;
        	   case SQL3_INTEGER:
        	   		printf("integer ");
        	   		break;
        	   case SQL3_SMALLINT:
        	   		printf("smallint ");
        	        break;
        	   case SQL3_FLOAT:
        	   		printf("float(%d,%d) ",PRECISION,SCALE);
        	        break;
        	   case SQL3_REAL:
        	   		printf("real ");
        	        break;
        	   case SQL3_DOUBLE_PRECISION:
        	   		printf("double precision ");
        	        break;
        	   case SQL3_DATE_TIME_TIMESTAMP:
        	   		{ ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_di_code,
	ECPGt_int,&(DATETIME_INTERVAL_CODE),(long)1,(long)1,sizeof(int), ECPGd_EODT);

#line 95 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 95 "dyntest.pgc"

        	   		switch(DATETIME_INTERVAL_CODE)
        	   		{	case SQL3_DDT_DATE:
        	   				printf("date "); break;
        	   			case SQL3_DDT_TIME:
        	   				printf("time "); break;
        	   			case SQL3_DDT_TIMESTAMP:
        	   				printf("timestamp "); break;
        	   			case SQL3_DDT_TIME_WITH_TIME_ZONE:
        	   				printf("time with time zone "); break;
        	   			case SQL3_DDT_TIMESTAMP_WITH_TIME_ZONE:
        	   				printf("timestamp with time zone "); break;
        	   		}
        	        break;
        	   case SQL3_INTERVAL:
        	   		printf("interval ");
        	        break;
        	   case SQL3_CHARACTER:
        	        if (LENGTH>0) printf("char(%d) ",LENGTH);
        	        else printf("char(?) ");
        	        break;
        	   case SQL3_CHARACTER_VARYING:
        	        if (LENGTH>0) printf("varchar(%d) ",LENGTH);
        	        else printf("varchar() ");
        	        break;
        	   default:
        	        if (TYPE<0) printf("<OID %d> ",-TYPE);
        	        else printf("<SQL3 %d> ",TYPE);
        	        break;
        	}
        	if (OCTET_LENGTH>0) printf("[%d bytes]",OCTET_LENGTH);
        	putchar('\n');
        }
        putchar('\n');
        done=1;
     }

     for (INDEX=1;INDEX<=COUNT;++INDEX)
     {
     	{ ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_indicator,
	ECPGt_int,&(INDICATOR),(long)1,(long)1,sizeof(int), ECPGd_precision,
	ECPGt_int,&(PRECISION),(long)1,(long)1,sizeof(int), ECPGd_scale,
	ECPGt_int,&(SCALE),(long)1,(long)1,sizeof(int), ECPGd_type,
	ECPGt_int,&(TYPE),(long)1,(long)1,sizeof(int), ECPGd_EODT);

#line 136 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 136 "dyntest.pgc"

     	if (INDICATOR==-1) printf("NULL");
        else switch (TYPE)
     	{	case SQL3_BOOLEAN:
     	        { ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_data,
	ECPGt_bool,&(BOOLVAR),(long)1,(long)1,sizeof(bool), ECPGd_EODT);

#line 140 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 140 "dyntest.pgc"

     			printf(BOOLVAR?"true":"false");
     			break;
     	   	case SQL3_NUMERIC:
     	   	case SQL3_DECIMAL:
     	        if (SCALE==0) /* we might even print leading zeros "%0*d" */
     	        {  { ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_data,
	ECPGt_int,&(INTVAR),(long)1,(long)1,sizeof(int), ECPGd_EODT);

#line 146 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 146 "dyntest.pgc"

     	           printf("%*d",PRECISION,INTVAR);
     	        }
     	        else
     	        {  { ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_data,
	ECPGt_float,&(FLOATVAR),(long)1,(long)1,sizeof(float), ECPGd_EODT);

#line 150 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 150 "dyntest.pgc"

     	           printf("%*.*f",PRECISION+1,SCALE,FLOATVAR);
     	        }
     	        break;
     	   	case SQL3_INTEGER:
     	   	case SQL3_SMALLINT:
     	        { ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_data,
	ECPGt_int,&(INTVAR),(long)1,(long)1,sizeof(int), ECPGd_EODT);

#line 156 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 156 "dyntest.pgc"

     	        printf("%d",INTVAR);
     	        break;
     	   	case SQL3_FLOAT:
     	   	case SQL3_REAL:
     	        { ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_data,
	ECPGt_float,&(FLOATVAR),(long)1,(long)1,sizeof(float), ECPGd_EODT);

#line 161 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 161 "dyntest.pgc"

     	        printf("%f",FLOATVAR);
     	        break;
     	   	case SQL3_DOUBLE_PRECISION:
     	        { ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_data,
	ECPGt_double,&(DOUBLEVAR),(long)1,(long)1,sizeof(double), ECPGd_EODT);

#line 165 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 165 "dyntest.pgc"

     	        printf("%f",DOUBLEVAR);
     	        break;
     	   	case SQL3_DATE_TIME_TIMESTAMP:
     	   	case SQL3_INTERVAL:
     	   	case SQL3_CHARACTER:
     	   	case SQL3_CHARACTER_VARYING:
     	   	default:
     	        { ECPGget_desc(__LINE__, "MYDESC", INDEX,ECPGd_data,
	ECPGt_char,(STRINGVAR),(long)1024,(long)1,(1024)*sizeof(char), ECPGd_EODT);

#line 173 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 173 "dyntest.pgc"

     	        printf("'%s'",STRINGVAR);
     	        break;
     	}
     	putchar('|');
     }
     putchar('\n');
  }

  { ECPGdo(__LINE__, 0, 1, NULL, "close MYCURS", ECPGt_EOIT, ECPGt_EORT);
#line 182 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );}
#line 182 "dyntest.pgc"

  ECPGdeallocate_desc(__LINE__, "MYDESC");
#line 183 "dyntest.pgc"

if (sqlca.sqlcode < 0) error (  );
#line 183 "dyntest.pgc"


  /* no exec sql disconnect; */

  return 0;
}
