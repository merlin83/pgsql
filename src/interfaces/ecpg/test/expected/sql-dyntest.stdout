6 Columns
rulename <OID 19> [64 bytes]
ev_class <OID 26> [4 bytes]
ev_attr smallint [2 bytes]
ev_type <OID 18> [1 bytes]
is_instead bool [1 bytes]
ev_qual char(?) 

'_RETURN'|'10297'|-1|'1'|true|'<>'|
'_RETURN'|'10300'|-1|'1'|true|'<>'|
'_RETURN'|'10303'|-1|'1'|true|'<>'|
'_RETURN'|'10306'|-1|'1'|true|'<>'|
'_RETURN'|'10309'|-1|'1'|true|'<>'|
'_RETURN'|'10313'|-1|'1'|true|'<>'|
'_RETURN'|'10316'|-1|'1'|true|'<>'|
'_RETURN'|'10320'|-1|'1'|true|'<>'|
'_RETURN'|'10324'|-1|'1'|true|'<>'|
'_RETURN'|'10328'|-1|'1'|true|'<>'|
'_RETURN'|'10331'|-1|'1'|true|'<>'|
'_RETURN'|'10334'|-1|'1'|true|'<>'|
'_RETURN'|'10337'|-1|'1'|true|'<>'|
'_RETURN'|'10340'|-1|'1'|true|'<>'|
'pg_settings_u'|'10340'|-1|'2'|false|'{OPEXPR :opno 98 :opfuncid 0 :opresulttype 16 :opretset false :args ({VAR :varno 2 :varattno 1 :vartype 25 :vartypmod -1 :varlevelsup 0 :varnoold 2 :varoattno 1} {VAR :varno 1 :varattno 1 :vartype 25 :vartypmod -1 :varlevelsup 0 :varnoold 1 :varoattno 1})}'|
'pg_settings_n'|'10340'|-1|'2'|true|'<>'|
'_RETURN'|'10345'|-1|'1'|true|'<>'|
'_RETURN'|'10348'|-1|'1'|true|'<>'|
'_RETURN'|'10352'|-1|'1'|true|'<>'|
'_RETURN'|'10355'|-1|'1'|true|'<>'|
'_RETURN'|'10358'|-1|'1'|true|'<>'|
'_RETURN'|'10362'|-1|'1'|true|'<>'|
'_RETURN'|'10365'|-1|'1'|true|'<>'|
'_RETURN'|'10368'|-1|'1'|true|'<>'|
'_RETURN'|'10372'|-1|'1'|true|'<>'|
'_RETURN'|'10375'|-1|'1'|true|'<>'|
'_RETURN'|'10378'|-1|'1'|true|'<>'|
'_RETURN'|'10382'|-1|'1'|true|'<>'|
'_RETURN'|'10385'|-1|'1'|true|'<>'|
'_RETURN'|'10388'|-1|'1'|true|'<>'|
'_RETURN'|'10391'|-1|'1'|true|'<>'|
'_RETURN'|'10394'|-1|'1'|true|'<>'|
'_RETURN'|'10397'|-1|'1'|true|'<>'|
'_RETURN'|'10400'|-1|'1'|true|'<>'|
'_RETURN'|'10634'|-1|'1'|true|'<>'|
'_RETURN'|'10638'|-1|'1'|true|'<>'|
'_RETURN'|'10641'|-1|'1'|true|'<>'|
'_RETURN'|'10644'|-1|'1'|true|'<>'|
'_RETURN'|'10648'|-1|'1'|true|'<>'|
'_RETURN'|'10651'|-1|'1'|true|'<>'|
'_RETURN'|'10655'|-1|'1'|true|'<>'|
'_RETURN'|'10659'|-1|'1'|true|'<>'|
'_RETURN'|'10663'|-1|'1'|true|'<>'|
'_RETURN'|'10667'|-1|'1'|true|'<>'|
'_RETURN'|'10671'|-1|'1'|true|'<>'|
'_RETURN'|'10675'|-1|'1'|true|'<>'|
'_RETURN'|'10678'|-1|'1'|true|'<>'|
'_RETURN'|'10681'|-1|'1'|true|'<>'|
'_RETURN'|'10684'|-1|'1'|true|'<>'|
'_RETURN'|'10688'|-1|'1'|true|'<>'|
'_RETURN'|'10691'|-1|'1'|true|'<>'|
'_RETURN'|'10695'|-1|'1'|true|'<>'|
'_RETURN'|'10699'|-1|'1'|true|'<>'|
'_RETURN'|'10703'|-1|'1'|true|'<>'|
'_RETURN'|'10707'|-1|'1'|true|'<>'|
'_RETURN'|'10711'|-1|'1'|true|'<>'|
'_RETURN'|'10715'|-1|'1'|true|'<>'|
'_RETURN'|'10718'|-1|'1'|true|'<>'|
'_RETURN'|'10722'|-1|'1'|true|'<>'|
'_RETURN'|'10726'|-1|'1'|true|'<>'|
'_RETURN'|'10729'|-1|'1'|true|'<>'|
'_RETURN'|'10767'|-1|'1'|true|'<>'|
'_RETURN'|'10771'|-1|'1'|true|'<>'|
'_RETURN'|'10775'|-1|'1'|true|'<>'|
'_RETURN'|'10779'|-1|'1'|true|'<>'|
'_RETURN'|'10782'|-1|'1'|true|'<>'|
'_RETURN'|'10786'|-1|'1'|true|'<>'|
'_RETURN'|'10789'|-1|'1'|true|'<>'|
'_RETURN'|'10793'|-1|'1'|true|'<>'|
'_RETURN'|'10797'|-1|'1'|true|'<>'|
'_RETURN'|'10801'|-1|'1'|true|'<>'|
'_RETURN'|'10805'|-1|'1'|true|'<>'|
'_RETURN'|'10809'|-1|'1'|true|'<>'|
